worker-policy-data-reprocessing-tracking/src/main/java/com/github/cafdataprocessing/worker/policy/data/reprocessing/tracking/TrackedDocument.java

```diff
@@ -134,7 +138,41 @@ public DocumentProcessingRecord createPolicyDataProcessingRecord() {
     public void deletePolicyDataProcessingRecord() {
         trackedDocumentChanger.getInternalDocument().deletePolicyDataProcessingRecord();
     }
-    
-   
-    
+
+    @Override
+    public Collection<DocumentInterface> getSubDocuments()
+    {
+        if (trackedSubDocuments == null) {
+            Collection<DocumentInterface> trackedsubDocumentsCollection = new ArrayList<>();
+            trackedDocumentChanger.getInternalDocument().getSubDocuments().forEach((document) -> {
+                trackedsubDocumentsCollection.add(new TrackedDocument(document));
+            });
+            this.trackedSubDocuments = trackedsubDocumentsCollection;
+        }
+        return Collections.unmodifiableCollection(trackedSubDocuments);
+    }
+
+    @Override
+    public void removeSubdocument(final String reference)
+    {
+        trackedDocumentChanger.getInternalDocument().getSubDocuments().stream().forEach(doc -> {
+            if (doc.getReference().equals(reference)) {
+                trackedDocumentChanger.getInternalDocument().removeSubdocument(reference);
```

Comment:
Should you exit the forEach once you have removed the subdocument requested. unless you think 2 docs with same reference to be removed, and if so should it be removeSubdocuments.